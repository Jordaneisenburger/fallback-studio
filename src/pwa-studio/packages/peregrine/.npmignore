.eslintrc.js
/lib/**/__{docs,helpers,mocks,tests}__/**
/scripts
