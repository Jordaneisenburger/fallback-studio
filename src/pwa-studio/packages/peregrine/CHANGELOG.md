# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.5.0"></a>
# 0.5.0 (2018-08-13)


### Features

* multicasting REST API client ([#164](https://github.com/magento/pwa-studio/issues/164)) ([2852e14](https://github.com/magento/pwa-studio/commit/2852e14)), closes [#140](https://github.com/magento/pwa-studio/issues/140)
* Simulators for async testing ([#24](https://github.com/magento/pwa-studio/issues/24)) ([a380da9](https://github.com/magento/pwa-studio/commit/a380da9))
