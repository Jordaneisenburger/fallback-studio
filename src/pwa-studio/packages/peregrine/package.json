{
  "name": "@magento/peregrine",
  "version": "2.1.0",
  "publishConfig": {
    "access": "public"
  },
  "description": "The core runtime of Magento PWA",
  "main": "dist/index.js",
  "scripts": {
    "build": "concurrently --raw yarn:build:cjs yarn:build:esm",
    "build:cjs": "BABEL_ENV=production babel src --out-dir dist --root-mode 'upward' --source-maps",
    "build:dev": "yarn run build",
    "build:esm": "BABEL_ENV=development babel src --out-dir esm --root-mode 'upward' --source-maps",
    "clean": "rimraf dist esm",
    "prepublishOnly": "yarn run clean && yarn run build",
    "storybook": "start-storybook -p 9001 -c .storybook",
    "storybook:build": "build-storybook -c .storybook -o storybook-dist",
    "watch": "yarn run build --watch"
  },
  "repository": "github:magento-research/pwa-studio",
  "author": "Magento Commerce",
  "license": "(OSL-3.0 OR AFL-3.0)",
  "bugs": {
    "url": "https://github.com/magento-research/pwa-studio/issues"
  },
  "homepage": "https://github.com/magento-research/pwa-studio/tree/master/packages/peregrine#readme",
  "devDependencies": {
    "@babel/cli": "~7.2.3",
    "@babel/core": "~7.3.4",
    "@babel/plugin-proposal-class-properties": "~7.3.4",
    "@babel/plugin-proposal-object-rest-spread": "~7.3.4",
    "@babel/plugin-syntax-dynamic-import": "~7.2.0",
    "@babel/plugin-syntax-jsx": "~7.2.0",
    "@babel/plugin-transform-react-jsx": "~7.3.0",
    "@babel/plugin-transform-runtime": "~7.4.4",
    "@babel/runtime": "~7.4.2",
    "@storybook/react": "~4.1.13",
    "concurrently": "~4.1.0",
    "intl": "~1.2.5",
    "react": "~16.8.3",
    "react-apollo": "~2.5.3",
    "react-dom": "~16.8.3",
    "react-router-dom": "~5.0.0",
    "react-test-renderer": "~16.8.6",
    "rimraf": "~2.6.3"
  },
  "peerDependencies": {
    "@babel/runtime": "^7.4.2",
    "react": "^16.8.0",
    "react-apollo": "^2.5.3",
    "react-dom": "^16.8.0",
    "react-router-dom": "~5.0.0"
  },
  "engines": {
    "node": "8.10.0 || >=10.7.0",
    "yarn": ">=1.12.0"
  },
  "module": "esm/index.js",
  "jsnext:main": "src/index.js",
  "es2015": "src/index.js",
  "sideEffects": false
}
