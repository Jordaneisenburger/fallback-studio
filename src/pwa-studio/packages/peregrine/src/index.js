import * as RestApi from './RestApi';
import * as Util from './util';

// hooks
export { useApolloContext } from './hooks/useApolloContext';
export { useEventListener } from './hooks/useEventListener';
export { useDropdown } from './hooks/useDropdown';
export { useQuery } from './hooks/useQuery';
export { useQueryResult } from './hooks/useQueryResult';
export { useSearchParam } from './hooks/useSearchParam';
export {
    WindowSizeContextProvider,
    useWindowSize
} from './hooks/useWindowSize';

// components
export { default as ContainerChild } from './ContainerChild';
export { default as List, Items, Item } from './List';
export { default as Page } from './Page';
export { default as Price } from './Price';
export { default as Router } from './Router';

// misc
export { RestApi };
export { Util };
export { default as createTestInstance } from './util/createTestInstance';
