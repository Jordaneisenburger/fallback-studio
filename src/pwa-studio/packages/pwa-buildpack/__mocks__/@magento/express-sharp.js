const middleware = jest.fn();
const expressSharp = (module.exports = jest.fn().mockReturnValue(middleware));
expressSharp.__mockMiddleware = middleware;
