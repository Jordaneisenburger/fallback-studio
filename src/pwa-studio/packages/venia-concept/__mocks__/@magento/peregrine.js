import React from 'react';
import { BrowserPersistence } from './util/simplePersistence';

const peregrine = jest.requireActual('../../../peregrine/lib');

// re-exports
const RestApi = {
    Magento2: {
        request: jest.fn()
    }
};

const Util = { BrowserPersistence };

// hooks
const useApolloContext = jest.fn(peregrine.useApolloContext);
const useEventListener = jest.fn(peregrine.useEventListener);
const useDropdown = jest.fn(peregrine.useDropdown);
const useQuery = jest.fn(peregrine.useQuery);
const useQueryResult = jest.fn(peregrine.useQueryResult);
const useSearchParam = jest.fn(peregrine.useSearchParam);
const useWindowSize = jest.fn(peregrine.useWindowSize);
const useToasts = jest.fn(peregrine.useToasts);
// components

/**
 * the Price component from @magento/peregrine
 * has browser-specific functionality and cannot
 * currently by rendered in the test environment
 */
const Price = jest.fn().mockReturnValue(<div />);

module.exports = {
    ...peregrine,
    Price,
    RestApi,
    Util,
    useApolloContext,
    useEventListener,
    useDropdown,
    useQuery,
    useQueryResult,
    useSearchParam,
    useWindowSize,

    useToasts
};
