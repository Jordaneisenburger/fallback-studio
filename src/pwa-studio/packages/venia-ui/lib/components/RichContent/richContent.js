import React from 'react';
import detectPageBuilder from './PageBuilder/detectPageBuilder';
import PageBuilder from './PageBuilder';
import { mergeClasses } from '../../classify';
import defaultClasses from './richContent.css';
import { shape, string } from 'prop-types';

const toHTML = str => ({ __html: str });

/**
 * RichContent component.
 *
 * This component serves as the pool to determine which type of content is being rendered
 * and pass the data off to the correct system.
 *
 * @typedef RichContent
 * @kind functional component
 *
 * @param {Object} props React component props
 *
 * @returns {React.Element} A React component that renders Heading with optional styling properties.
 */
const RichContent = props => {
    const { html } = props;
    const classes = mergeClasses(defaultClasses, props.classes);

    if (detectPageBuilder(html)) {
        return (
            <div className={classes.root}>
                <PageBuilder masterFormat={html} />
            </div>
        );
    }

    return (
        <div className={classes.root} dangerouslySetInnerHTML={toHTML(html)} />
    );
};

/**
 * Props for {@link RichContent}
 *
 * @typedef props
 *
 * @property {Object} classes An object containing the class names for the RichContent
 * @property {String} classes.root CSS class for the root container element
 * @property {String} html Content
 */
RichContent.propTypes = {
    classes: shape({
        root: string
    }),
    html: string
};

export default RichContent;
