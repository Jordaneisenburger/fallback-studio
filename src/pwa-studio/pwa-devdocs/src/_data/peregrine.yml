title: Peregrine
entries:
    - label: Overview
      url: /peregrine/

    - label: Routing in PWA Studio
      url: /peregrine/routing/

    - label: Utility modules
      entries:
          - label: REST API client
            url: /peregrine/reference/rest-api-client/

    - label: Custom React hooks
      entries:
          - label: useDropdown
            url: /peregrine/reference/hooks/useDropdown/

          - label: useEventListener
            url: /peregrine/reference/hooks/useEventListener/

          - label: usePagination
            url: /peregrine/reference/hooks/usePagination/

          - label: useSearchParam
            url: /peregrine/reference/hooks/useSearchParam/

          - label: useToasts
            url: /peregrine/reference/toasts/useToasts

          - label: useToastContext
            url: /peregrine/reference/toasts/useToastContext/

          - label: useWindowSize
            url: /peregrine/reference/hooks/useWindowSize/

    - label: React components
      entries:
          - label: ContainerChild
            url: /peregrine/reference/container-child/

          - label: Item
            url: /peregrine/reference/item/

          - label: Items
            url: /peregrine/reference/items/

          - label: List
            url: /peregrine/reference/list/

          - label: Price
            url: /peregrine/reference/price/

          - label: Router
            url: /peregrine/reference/router/
