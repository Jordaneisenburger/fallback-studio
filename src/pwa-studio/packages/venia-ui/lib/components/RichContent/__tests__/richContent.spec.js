import React from 'react';
import { createTestInstance } from '@magento/peregrine';
import RichContent from '../richContent';

jest.mock('../PageBuilder/config', () => {
    return () => {
        return {
            configAggregator: () => {},
            component: ({ contentType, children }) => (
                <div dataContentType={contentType}>{children}</div>
            )
        };
    };
});

test('renders a RichContent component', () => {
    const component = createTestInstance(<RichContent />);

    expect(component.toJSON()).toMatchSnapshot();
});

test('renders a RichContent component with Page Builder content', () => {
    const richContentProps = {
        html:
            '<div data-content-type="row" data-appearance="contained" data-element="main"><div data-enable-parallax="0" data-parallax-speed="0.5" data-background-images="{}" data-element="inner" style="justify-content: flex-start; display: flex; flex-direction: column; background-position: left top; background-size: cover; background-repeat: no-repeat; background-attachment: scroll; border-style: solid; border-color: rgb(255, 0, 0); border-width: 5px; border-radius: 2px; margin: 0px 0px 10px; padding: 10px;"></div></div>'
    };
    const component = createTestInstance(<RichContent {...richContentProps} />);

    expect(component.toJSON()).toMatchSnapshot();
});

test('renders a RichContent component with HTML content', () => {
    const richContentProps = {
        html: '<h1>Testing HTML content</h1>'
    };
    const component = createTestInstance(<RichContent {...richContentProps} />);

    expect(component.toJSON()).toMatchSnapshot();
});
