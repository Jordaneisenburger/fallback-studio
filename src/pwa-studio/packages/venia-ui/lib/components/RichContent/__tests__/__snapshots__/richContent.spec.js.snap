// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a RichContent component 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": undefined,
    }
  }
/>
`;

exports[`renders a RichContent component with HTML content 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<h1>Testing HTML content</h1>",
    }
  }
/>
`;

exports[`renders a RichContent component with Page Builder content 1`] = `
<div>
  <div
    dataContentType="row"
  />
</div>
`;
